src/wazuh_db/wdb_delta_event.c

```diff
@@ -1,425 +1,273 @@
+                if (column->value.is_pk) {
+                    const char * field_name = wdb_dbsync_translate_field(&column->value);
+                    cJSON * field_value = cJSON_GetObjectItem(data, field_name);
+                    if (NULL != field_value &&
+                        !wdb_dbsync_stmt_bind_from_json(stmt, index, column->value.type, field_value,
+                                                        column->value.can_be_null)) {
+                        merror(DB_INVALID_DELTA_MSG, wdb->id, field_name, kv_value->key);
+                        has_error = true;
                     }
+                    ++index;
                 }
             }
-            ret_val = !has_error && SQLITE_DONE == wdb_step(stmt) && sqlite3_changes(wdb->db) > 0;
-            os_free(data_temp);
+            ret_val = !has_error && SQLITE_DONE == wdb_step(stmt);
         } else {
             merror(DB_CACHE_NULL_STMT);
         }
     }
     return ret_val;
 }
 
+STATIC bool wdb_dbsync_stmt_bind_from_json(sqlite3_stmt * stmt, int index, field_type_t type, const cJSON * value,
+                                           bool can_be_null) {
 
-void wdb_select_dbsync(wdb_t * wdb, struct kv const *kv_value, const char *data, char *output)
-{
-    if (NULL != wdb && NULL != data) {
-        char query[OS_SIZE_2048] = { 0 };
-        bool first_condition_element = true;
-        struct column_list const *column = NULL;
-
-        strcat(query, "SELECT ");
-        for (column = kv_value->column_list; column ; column=column->next) {
-            if (!column->value.is_old_implementation) {
-                if (first_condition_element) {
-                    first_condition_element = false;
-                } else {
-                    strcat(query, ", ");
-                }
-                strcat(query, column->value.name);
-            }
-        }
-        strcat(query, " FROM ");
-        strcat(query, kv_value->value);
-        strcat(query, " WHERE ");
+    bool ret_val = false;
 
-        first_condition_element = true;
-        for (column = kv_value->column_list; column ; column=column->next) {
-            if (!column->value.is_old_implementation) {
-                if (column->value.is_pk) {
-                    if (first_condition_element) {
-                        strcat(query, column->value.name);
-                        strcat(query, "=?");
-                        first_condition_element = false;
+    if (NULL != stmt && NULL != value) {
+        if (cJSON_NULL == value->type) {
+            ret_val = sqlite3_bind_null(stmt, index) == SQLITE_OK ? true : false;
+        } else {
+            switch (type) {
+            case FIELD_TEXT: {
+                switch (value->type) {
+                case cJSON_String:
+                    if ('\0' == *value->valuestring && can_be_null) {
+                        ret_val = sqlite3_bind_null(stmt, index) == SQLITE_OK ? true : false;
```

Comment:
Basically `can_be_null` allow treating as `NULL` those strings that are empty. This field is important for those fields that are vuln-related, like 'cpe' that need an empty string rather than a null.

Reply:
Sorry, I made a mistake. I was talking about the line 190.

Reply:
can_be_null is only to strings that should be inserted into the DB. Values that are explicitly `null` in the JSON (line 190) are treated as db's null